staging/src/k8s.io/apiextensions-apiserver/test/integration/ratcheting_test.go

```diff
@@ -95,6 +97,12 @@ type expectError struct {
 func (e expectError) Do(ctx *ratchetingTestContext) error {
 	err := e.op.Do(ctx)
 	if err != nil {
+		// If there is an error, it should happen when updating the CR
+		// and also when updating the CR status subresource.
+		if agg, ok := err.(utilerrors.Aggregate); ok && len(agg.Errors()) != 2 {
+			return fmt.Errorf("expected 2 errors, got %v", len(agg.Errors()))
+		}
```

Comment:
this extra logic has some smell. `expectError` does not sound like it will check that there are two errors aggregated.